lib/encoder.js

```diff
@@ -890,6 +907,95 @@ function defineInstanceMembers() {
     return Buffer.concat(parts, totalLength);
   };
 
+  this.decodeVector = function(buffer, params) {
+    const subtype = params["subtype"];
+    const dimensions = params["dimensions"];
+    const elemLength = 4; // TODO: figure this out based on the subtype
```

Comment:
I think we want to explicitly support only float subtypes since that's what cassandra does? What do you do in the other drivers @absurdfarce ?

Reply:
We're only supporting floats _for now_ @phact ... that's why I have some of the error messaging says what it says at the moment.  But there's no guarantee that it'll stay that way going forward and I don't really wanna build that assumption into the code.  We should be able to determine a size pretty readily based on subtype; we can then just add new subtypes as necessary in the future as they're added to Cassandra.

Re: the other drivers... the Java driver [does something really dumb](https://github.com/datastax/java-driver/blob/4.x/core/src/main/java/com/datastax/oss/driver/internal/core/type/codec/VectorCodec.java#L115-L126) to handle this.  Fixing that is _very_ high on my list of "things I'd like to do with the driver in the future".  The Python driver does [something similar to what we're doing here](https://github.com/datastax/python-driver/blob/master/cassandra/cqltypes.py#L1444), again as a temporary expedient knowing that the current release only supports 4 byte floats.

Eventually all of these impls should be generalized so that we don't have this hidden assumption about data size built-in.  There's a _separate_ concern about the fact that we only support fixed-size subtypes so anything that has a variable size will definitely break things.  There was some conversation around this on [CASSANDRA-18504](https://issues.apache.org/jira/browse/CASSANDRA-18504) but I wouldn't say it's resolved.